update documentation for managed service migrations using migration assistant (#9338)
opensearch-project/documentation-website: _migration-assistant/migration-phases/planning-your-migration/verifying-migration-tools.md

#### Access denied error (403)

> If you encounter an error like `AccessDenied (Service: Amazon S3; Status Code: 403)`, verify that:

Reviewer: ```suggestion
If you encounter an error like `AccessDenied (Service: Amazon S3; Status Code: 403)`, verify the following:
```

> - You're using the S3 bucket created by Migration Assistant (recommended)

Reviewer: ```suggestion
- Make sure you're using the S3 bucket created by Migration Assistant.
```

> - If using a custom S3 bucket, verify that:

Reviewer: ```suggestion
- If you're using a custom S3 bucket, verify that:
```

>   - The IAM role assigned to your Elasticsearch cluster has the necessary S3 permissions

Reviewer: ```suggestion
  - The IAM role assigned to your Elasticsearch cluster has the necessary S3 permissions.
```

>   - The bucket name and AWS Region provided in the snapshot configuration match the actual S3 bucket you created

Reviewer: ```suggestion
  - The bucket name and AWS Region provided in the snapshot configuration match the actual S3 bucket you created.
```

Look for failing tasks by navigating to **Traffic Capture Proxy ECS**. Change **Filter desired status** to **Any desired status** in order to see all tasks and navigate to the logs for stopped tasks.

### Snapshot and S3 bucket issues

> When using the CDK deployment for Migration Assistant, you might encounter the following errors duing snapshot creation and deletion:

Reviewer: ```suggestion
When using the CDK deployment for Migration Assistant, you might encounter the following errors during snapshot creation and deletion:
```

#### Bucket permissions

To make sure you can delete snapshots as well as create them during the CDK deployment process, confirm that `OSMigrations-dev-<region>-CustomS3AutoDeleteObjects` stack has S3 object deletion rights. Then, verify that `OSMigrations-dev-<region>-default-SnapshotRole` has the following S3 permissions:  

  - List bucket contents  
  - Read/Write/Delete objects

#### Snapshot conflicts

> To prevent snapshot conflicts, use `console snapshot delete` command from the migration console. If you delete snapshots or snapshot repositories in a place other than the migration consolve, you could encounter "already exists" errors.

Reviewer: ```suggestion
To prevent snapshot conflicts, use `console snapshot delete` command from the migration console. If you delete snapshots or snapshot repositories in a place other than the migration console, you could encounter "already exists" errors.
```

Look for failing tasks by navigating to **Traffic Capture Proxy ECS**. Change **Filter desired status** to **Any desired status** in order to see all tasks and navigate to the logs for stopped tasks.

### Snapshot and S3 bucket issues

> When using the CDK deployment for Migration Assistant, you might encounter the following errors during snapshot creation and deletion:

Reviewer 2: ```suggestion
When using the CDK deployment for Migration Assistant, you might encounter the following errors during snapshot creation and deletion.
```

#### Bucket permissions

> To make sure you can delete snapshots as well as create them during the CDK deployment process, confirm that `OSMigrations-dev-<region>-CustomS3AutoDeleteObjects` stack has S3 object deletion rights. Then, verify that `OSMigrations-dev-<region>-default-SnapshotRole` has the following S3 permissions:  

Reviewer 2: ```suggestion
To make sure that you can delete snapshots as well as create them during the CDK deployment process, confirm that the `OSMigrations-dev-<region>-CustomS3AutoDeleteObjects` stack has S3 object deletion rights. Then, verify that `OSMigrations-dev-<region>-default-SnapshotRole` has the following S3 permissions:  
```

  - List bucket contents  
  - Read/Write/Delete objects

#### Snapshot conflicts

> To prevent snapshot conflicts, use `console snapshot delete` command from the migration console. If you delete snapshots or snapshot repositories in a place other than the migration console, you could encounter "already exists" errors.

Reviewer 2: ```suggestion
To prevent snapshot conflicts, use the `console snapshot delete` command from the migration console. If you delete snapshots or snapshot repositories in a location other than the migration console, you might encounter "already exists" errors.
```

The following steps outline how to reset resources with Migration Assistant before executing the actual migration. At this point all verifications are expected to have been completed. These steps can be performed after [Accessing the Migration Console]({{site.url}}{{site.baseurl}}/migration-assistant/migration-console/accessing-the-migration-console/).

### Snapshot and S3 Bucket Issues

1. **Bucket Permissions**  

>   Confirm the `OSMigrations-dev-<region>-CustomS3AutoDeleteObjects` stack has S3 object deletion rights.

Reviewer 3: ? Why is this needed, shouldn't this already be done with the CDK setup assuming you use the S3 bucket that MA created?

Author: Yes you are right. Let me update this section for users who do not use the S3 bucket created by MA.

Reviewer 3: Still doesn't seem like the right place for this information. When you are using the MA bucket these steps shouldn't be followed - if we had an issue it would be a bug we'd fix, as opposed to adding a section in troubleshooting docs.

It seems like a new doc on how to use your own S3 bucket (byob- hah) would be a better "success" driven scenario. What do you think about the scenarios placement and approach?

The following steps outline how to reset resources with Migration Assistant before executing the actual migration. At this point all verifications are expected to have been completed. These steps can be performed after [Accessing the Migration Console]({{site.url}}{{site.baseurl}}/migration-assistant/migration-console/accessing-the-migration-console/).

> ### Snapshot and S3 bucket issues

Reviewer: @jugal-chauhan: What are the issues these steps are trying to solve? "Bucket permissions" and "Migration role access" aren't quite specific enough.

Author: These steps are guidelines to troubleshoot errors during Snapshot Creation when using the CDK Deployment of Migration Assistant. Users can delete snapshots, but if they deleted the Snapshot Repository by mistake then the new repository will face these 'migration role access' issues. I am calling out such mistakes in Step 3 - so that users do not delete the snapshot repository.

The most important thing to verify is that the S3 bucket registered by the MigrationBootstrap Stack during the CDK deployment process, has sufficient permissions which was the default behavior of the MigrationBootstrap Stack.
